TestStack.ConventionTests/ConventionData/Types.cs

```diff
@@ -1,49 +1,221 @@
 ﻿namespace TestStack.ConventionTests.ConventionData
 {
     using System;
+    using System.Collections;
     using System.Collections.Generic;
+    using System.ComponentModel;
     using System.Linq;
-    using System.Runtime.CompilerServices;
+    using System.Reflection;
 
     /// <summary>
     ///     This is where we set what our convention is all about.
     /// </summary>
-    public class Types : IConventionData
+    public class Types : IConventionData, IEnumerable<Type>
     {
-        public Types(string descriptionOfTypes)
+        private Types(IEnumerable<Type> typesToVerify, string descriptionOfTypes)
         {
+            TypesToVerify = typesToVerify;
             Description = descriptionOfTypes;
         }
 
-        public Type[] TypesToVerify { get; set; }
```

Comment:
I think we should at least .ToArray() internally. The reason I use array is so make sure that it is not a lazy enumerable which can be executed multiple times.